src/storage/hummock_sdk/src/lib.rs

```diff
@@ -266,3 +266,56 @@ pub fn version_checkpoint_path(root_dir: &str) -> String {
 pub fn version_checkpoint_dir(checkpoint_path: &str) -> String {
     checkpoint_path.trim_end_matches(|c| c != '/').to_string()
 }
+
+// const EPOCH_AVAILABLE_BITS: u64 = 16;
+const EPOCH_MASK: u64 = 0xFFFF;
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Default, Debug, PartialOrd, Ord)]
+pub struct EpochWithGap(u64);
+
+impl EpochWithGap {
+    pub fn new(epoch: u64, spill_offset: u64) -> Self {
+        #[cfg(not(feature = "enable_test_epoch"))]
+        {
+            let epoch_with_gap = epoch + spill_offset;
+            EpochWithGap(epoch_with_gap)
+        }
+        #[cfg(feature = "enable_test_epoch")]
+        {
+            let epoch_with_gap = epoch + spill_offset - spill_offset;
+            EpochWithGap(epoch_with_gap)
+        }
+    }
+
+    pub fn new_from_epoch(epoch_with_gap: u64) -> Self {
```

Comment:
Better to call `Self::new(epoch, 0)` to be consistent. And the parameter name changed to `epoch: u64`.